upstairs/src/dummy_downstairs_tests.rs

```diff
@@ -1473,6 +1528,17 @@ async fn test_byte_fault_condition() {
         assert_eq!(ds[ClientId::new(0)], DsState::Active);
         assert_eq!(ds[ClientId::new(1)], DsState::Active);
         assert_eq!(ds[ClientId::new(2)], DsState::Active);
+
+        // Once one of the Downstairs has cached more that a certain amount of
+        // bytes, it will automatically send a Barrier operation.
+        if i == barrier_point {
+            let m = harness.ds1().recv().await.unwrap();
+            if !matches!(m, Message::Barrier { .. }) {
+                panic!("expected Barrier, got message {m:?}");
+            }
+            harness.ds2.ack_barrier().await;
+            harness.ds3.ack_barrier().await;
```

Comment:
Do we have any way to check that replay is no longer available to this downstairs?  I'm not sure if the test has access at this layer to it.

Reply:
The end of this unit test is checking that live-repair works, so I think we're implicitly testing that we don't do replay.

Reply:
Ah, I was thinking more that the sending of a barrier operation has also flipped the `can_replay` in the `Downstairs` struct.  I'm not sure if we can do that though here (or in a test elsewhere), as `can_replay` might not be exposed like that.

Reply:
Yeah, it's not easy to check from the outside (when we only have the `Guest` handle).  If you really want it, we could probably add it to the `downstairs_state` helper function.

Reply:
I don't think it's worth adding it to the `downstairs_state` helper function. 
I don't see anywhere else that seems like a good place either.  Nothing in the upstairs tests cover this case.